prime-router/src/main/kotlin/azure/EmailEngineFunction.kt

```diff
@@ -0,0 +1,328 @@
+        var jwtToken = request.headers["authorization"] ?: ""
+
+        jwtToken = if (jwtToken.length > 7) jwtToken.substring(7) else ""
+
+        if (jwtToken.isNotBlank()) {
+            try {
+                // get the access token verifier
+                val jwtVerifier = JwtVerifiers.accessTokenVerifierBuilder()
+                    .setIssuer("https://hhs-prime.okta.com/oauth2/default")
+                    .build()
+                // get it to decode the token from the header
+                val jwt = jwtVerifier.decode(jwtToken)
+                    ?: throw Throwable("Error in validation of jwt token")
+                // get the user name and org
+                return jwt.claims["sub"].toString();
+            }
+            catch (ex: Throwable) {
+                context.logger.log(Level.WARNING, "Error in verification of token", ex)
+                return null
+            }
+        }
+        return null;
+    }
+    
+
+    /**
+     * TODO: Fixme!
+     */
+    private fun getSchedules(): List<String>{
+        return WorkflowEngine.databaseAccess.fetchEmailSchedules();
+    }
+
+    /**
+     * Converts an organization name to an OKTA group name
+     */
+    private fun convertOrgToGroup( org: String ): String {
+        return "DH"+org.replace( "-","_" )
+    }
+
+    /**
+     * Retrieve a list of emails within an organization
+     * 
+     * @params org The organization to fetch emails for
+     * @returns List of emails to send to
+     */
+    private fun getEmails( org: String, context: ExecutionContext ): List<String> {
+
+        var ssws: String? = SecretHelper.getSecretService().fetchSecret("SSWS_OKTA")
+
+        if( ssws == null ){
+            context.logger.warning("Can't find SSWS_OKTA secret")
+            return listOf();
+        }
+        
+        var grp = convertOrgToGroup( org );
+
+        // get the OKTA Group Id
+        var response1 = httpGet( url="https://hhs-prime-admin.okta.com/api/v1/groups?q=${grp}", 
+                               headers=mapOf( "Authorization" to "SSWS ${ssws}" ) );
+        var grpId = ((response1.jsonArray).get( 0 ) as JSONObject).getString("id");
+
+        // get the users within that OKTA group
+        var response = httpGet( url="https://hhs-prime-admin.okta.com/api/v1/groups/${grpId}/users", 
+                               headers=mapOf( "Authorization" to "SSWS ${ssws}" ) );
+    
+       var emails :MutableList<String> = mutableListOf()
+
+        for( user in response.jsonArray ){
+            emails.add( (user as JSONObject).getJSONObject("profile").getString("email") );
+        } 
+
+        return emails;
+    }
+
+     /**
+     * Get the reports that have been generated since a given date
+     * 
+     * @param org
+     * 
+     */
+    private fun getReportsSinceLast( org: String, last: Date ): List<ReportFile> {
+
+      val reportFiles = workflowEngine
```

Comment:
Nitpick: formatting; also consider just returning and not storing in the variable (for conciseness).

Reply:
Agree - done